contributors/devel/release.md

```diff
@@ -0,0 +1,231 @@
+Automation and the release team will be in contact with you if
+updates are required, but inaction can result in your work being
+removed from the milestone.  Additional requirements exist when the
+target milestone is a prior release (see [cherry pick
+process](cherry-picks.md) for more information).
+
+In the past there was a requirement for a milestone targeted pull
+request to have an associated GitHub issue opened, but this is no
+longer the case.  Features are effectively GitHub issues (or KEPs)
+which lead to subsequent PRs.  The terms feature, issue, and PR may
+be used interchangeably below, but the general labeling process
+should be consistent across artifact types.
+
+---
+
+## Definitions
+
+- *issue owners*: Creator, assignees, and user who moved the issue into a release milestone.
+- *release team*: Each kubernetes release has a team doing project
+  management tasks described
+  [here](https://git.k8s.io/sig-release/release-team/README.md).  The
+  contact info for the team associated with any given release can be
+  found [here](https://git.k8s.io/sig-release/releases/).
+- *Y days*: Refers to business days (using the location local to the release-manager M-F).
+
+## The Release Cycle
+
+![Image of one kubernetes release cycle](release-cycle.png)
+
+Kubernetes releases currently happen four times per year.  The release
+process can be thought of as having three main phases:
+* feature definition
+* implementation
+* stabilization
+But in reality this is an open source and agile project, with feature
+planning and implementation happening at all times.  And given the
+project scale and globally distributed developer base, it is critical
+to project velocity that continuous integration testing insures the
+project is continuous stable so that individual commits can be
+flagged as having broken something.
+
+With ongoing feature definition through the year, some set of items
+will bubble up as targeting a given release.  The **feature freeze**
+starts ~4 weeks into release cycle.  By this point all intended
+feature work for the given release has been defined in suitable
+planning artifacts in conjunction with the Release Team's [features
+lead](https://github.com/kubernetes/sig-release/tree/master/release-team/role-handbooks/features).
+
+Implementation and bugfixing is ongoing across the cycle, but
+culminates in a code slush and code freeze period:
+* The **code slush** starts in week ~9 of the release cycle.  The master
+  branch only accepts PRs for release milestone.  No additional feature
+  development is merged after this point.
+* The **code freeze** starts in week ~10 and continues for ~2 weeks.
+  Only critical bug fixes are accepted into the release codebase.
+
+After this point only critical bug fixes happen and all remaining
+issues are resolved before release.
+
+When the code base is sufficiently stable, the master branch re-opens for
+general development and the release is finalize from a branch.
```

Comment:
typo: "finalized"
clarify this is the end of code freeze and all merges to master will target the _next_ release? maybe I'm overly picky here